builtin/providers/vsphere/resource_vsphere_virtual_disk.go

```diff
@@ -0,0 +1,250 @@
+					value := v.(string)
+					if value != "thin" && value != "eagerZeroedThick" {
+						errors = append(errors, fmt.Errorf(
+							"only 'thin' and 'eagerZeroedThick' are supported values for 'type'"))
+					}
+					return
+				},
+			},
+
+			"adapter_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "ide",
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if value != "ide" && value != "busLogic" && value != "lsiLogic" {
+						errors = append(errors, fmt.Errorf(
+							"only 'ide', 'busLogic', and 'lsiLogic' are supported values for 'adapter_type'"))
+					}
+					return
+				},
+			},
+
+			"datacenter": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"datastore": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceVSphereVirtualDiskCreate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[INFO] Creating Virtual Disk")
+	client := meta.(*govmomi.Client)
+
+	vDisk := virtualDisk{
+		size: d.Get("size").(int),
+	}
+
+	if v, ok := d.GetOk("vmdk_path"); ok {
+		vDisk.vmdkPath = v.(string)
+	}
+
+	if v, ok := d.GetOk("type"); ok {
+		vDisk.initType = v.(string)
+	}
+
+	if v, ok := d.GetOk("adapter_type"); ok {
+		vDisk.adapterType = v.(string)
+	}
+
+	if v, ok := d.GetOk("datacenter"); ok {
+		vDisk.datacenter = v.(string)
+	}
+
+	if v, ok := d.GetOk("datastore"); ok {
+		vDisk.datastore = v.(string)
+	}
+
+	diskPath := fmt.Sprintf("[%v] %v", vDisk.datastore, vDisk.vmdkPath)
+
+	err := createHardDisk(client, vDisk.size, diskPath, vDisk.initType, vDisk.adapterType, vDisk.datacenter)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(diskPath)
+	log.Printf("[DEBUG] Virtual Disk id: %v", diskPath)
+
+	return resourceVSphereVirtualDiskRead(d, meta)
+}
+
+func resourceVSphereVirtualDiskRead(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] Reading virtual disk.")
+	client := meta.(*govmomi.Client)
+
+	vDisk := virtualDisk{
+		size: d.Get("size").(int),
+	}
+
+	if v, ok := d.GetOk("vmdk_path"); ok {
+		vDisk.vmdkPath = v.(string)
+	}
+
+	if v, ok := d.GetOk("type"); ok {
+		vDisk.initType = v.(string)
+	}
+
+	if v, ok := d.GetOk("adapter_type"); ok {
+		vDisk.adapterType = v.(string)
+	}
+
+	if v, ok := d.GetOk("datacenter"); ok {
+		vDisk.datacenter = v.(string)
+	}
+
+	if v, ok := d.GetOk("datastore"); ok {
+		vDisk.datastore = v.(string)
+	}
+
+	dc, err := getDatacenter(client, d.Get("datacenter").(string))
+	if err != nil {
+		return err
+	}
+
+	finder := find.NewFinder(client.Client, true)
+	finder = finder.SetDatacenter(dc)
+
+	ds, err := finder.Datastore(context.TODO(), d.Get("datastore").(string))
+	if err != nil {
+		return err
+	}
+
+	fileInfo, err := ds.Stat(context.TODO(), vDisk.vmdkPath)
```

Comment:
if you don't need fileInfo, you can replace it with:
_, err = ds.Stat(context.TODO(), vDisk.vmdkPath)